src/shogun/classifier/LDA.cpp

```diff
@@ -216,22 +238,32 @@ bool CLDA::train_machine(CFeatures *data)
 		// to find SVD((inverse(Chol(Sw)))' * Sb * (inverse(Chol(Sw))))
 		//1.get Cw=Chol(Sw)
 		//find the decomposition of Cw'.
-		HouseholderQR<MatrixXd> decomposition(Sw.llt().matrixU().transpose());
+		HouseholderQR<typename SGMatrix<ST>::EigenMatrixXt> decomposition(Sw.llt().matrixU().transpose());
 
 		//2.get P=inv(Cw')*Sb_new
 		//MatrixXd P=decomposition.solve(Sb);
 		//3. final value to be put in SVD will be therefore:
 		// final_ output =(inv(Cw')*(P'))'
-		JacobiSVD<MatrixXd> svd2(decomposition.solve((decomposition.solve(Sb))
+		JacobiSVD<typename SGMatrix<ST>::EigenMatrixXt> svd2(decomposition.solve((decomposition.solve(Sb))
 					.transpose()).transpose(), ComputeThinU);
 
 		// Since this is a linear classifier, with only binary classes,
 		// we need to keep only the 1st eigenvector.
-		Map<VectorXd> x(w.vector, num_feat);
+		Map<typename SGVector<ST>::EigenVectorXt> x(w_st.vector, num_feat);
 		x=Q*(svd2.matrixU().col(0));
 		// get the bias
-		bias=(x.transpose()*mean_total);
+		bias= ((float64_t) (x.transpose()*mean_total));
 		bias=bias*(-1);
 	}
+
+ 	w=SGVector<float64_t>(num_feat);
+	w.zero();
+
+	//copy w_st into w
+	for(i = 0; i < w.size(); ++i)
```

Comment:
I'll implement this in a separate PR, since it requires a class/struct which contains the transformation operation.  It would be a good utility method.